File: app/src/main/java/com/silentgames/silent_planet/model/entities/ground/PlayersOnCell.java
package com.silentgames.silent_planet.model.entities.ground;

import com.silentgames.silent_planet.model.entities.EntityType;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by gidroshvandel on 26.09.16.
 */
public class PlayersOnCell{
    List<Player> playerList = new ArrayList<>();

    public List<Player> getPlayerList() {
        return new ArrayList<>(playerList);
    }

    public void setPlayerList(List<Player> playerList) {
        this.playerList = playerList;
    }

    public void add(Player player) {
        if(playerList == null){
            playerList = new ArrayList<>();
            playerList.add(player);
        }else {
            playerList.add(player);
        }
    }

    public Player getPlayerByName(String name) {
        for (Player player: playerList
                ) {
            if(player.getPlayerName() == name){
                return player;
            }
        }
        return null;
    }

    public void removePlayerFromBoard(Player player) {
        if(playerList != null){
            playerList.remove(player);
        }
    }

    public void removePlayerByName(String name) {
        if(playerList.size() == 1){
            playerList = null;
        }else {
            for (Player player : playerList
                    ) {
                if (player.getPlayerName() == name) {
                    playerList.remove(player);
                    break;
                }
            }
        }
    }
}

File: app/src/main/java/com/silentgames/silent_planet/model/entities/ground/PlayersOnCell.kt
package com.silentgames.silent_planet.model.entities.ground

import com.silentgames.silent_planet.model.entities.EntityType

import java.util.ArrayList

/**
 * Created by gidroshvandel on 26.09.16.
 */
class PlayersOnCell {
    internal var playerList: MutableList<Player>? = ArrayList()

    fun getPlayerList(): List<Player>? {
        return playerList
    }

    fun setPlayerList(playerList: MutableList<Player>) {
        this.playerList = playerList
    }

    fun add(player: Player) {
        if (playerList == null) {
            playerList = ArrayList()
            playerList!!.add(player)
        } else {
            playerList!!.add(player)
        }
    }

    fun getPlayerByName(name: String): Player? {
        for (player in playerList!!) {
            if (player.playerName === name) {
                return player
            }
        }
        return null
    }

    fun removePlayerFromBoard(player: Player) {
        if (playerList != null) {
            playerList!!.remove(player)
        }
    }

    fun removePlayerByName(name: String) {
        if (playerList!!.size == 1) {
            playerList = null
        } else {
            for (player in playerList!!) {
                if (player.playerName === name) {
                    playerList!!.remove(player)
                    break
                }
            }
        }
    }
}
